pkg/diagnose/runner.go

```diff
@@ -240,43 +243,84 @@ func RunAll(diagCfg diagnosis.Config) []diagnosis.Diagnoses {
 	return suiteDiagnoses
 }
 
+func requestDiagnosesFromAgentProcess(diagCfg diagnosis.Config) []diagnosis.Diagnoses {
+	c := util.GetClient(false)
+	ipcAddress, err := pkgconfig.GetIPCAddress()
+	if err != nil {
+		fmt.Fprintln(color.Output, color.RedString(fmt.Sprintf("Error getting IPC address for the agent: %s, running diagnose locally", err)))
+		return getDiagnosesFromCurrentProcess(diagCfg)
+	}
+
+	// Set session token
+	if err = util.SetAuthToken(); err != nil {
+		fmt.Fprintln(color.Output, color.RedString(fmt.Sprintf("Auth error: %s, running diagnose locally", err)))
+		return getDiagnosesFromCurrentProcess(diagCfg)
```

Comment:
This function mixes reporting and actual request. This will make it hard to test, and there is a bit of code repetition.

Would it make sense to rewrite this to function to handle request part only, and return `([]Diagnoses, error)`, and handle reporting and the fallback to in-process at the caller site?